spec/ndx-ophys-devices.extensions.yaml

```diff
@@ -1,8 +1,216 @@
+      complex transmission spectra.
+    required: false
+  - name: reflection_bandwidth_in_nm
+    dtype: float
+    doc: The range of wavelengths that are primarily reflected.The start and end wavelengths
+      needs to be specified.
+    required: false
+  - name: transmission_bandwidth_in_nm
+    dtype: float
+    doc: The range of wavelengths that are primarily transmitted.The start and end
+      wavelengths needs to be specified.
+    required: false
+  - name: angle_of_incidence_in_degrees
+    dtype: float
+    doc: Intended angle at which light strikes the mirror.
+    required: false
+  - name: model
+    dtype: text
+    doc: Model of the dichroic mirror.
+    required: false
+- neurodata_type_def: BandOpticalFilter
+  neurodata_type_inc: Device
+  doc: Extends Device to hold a Band Optical Filter (Bandpass or Bandstop).
+  attributes:
+  - name: center_wavelength_in_nm
+    dtype: float
+    doc: The midpoint of the band of wavelengths that the filter transmits or blocks.
+  - name: bandwidth_in_nm
+    dtype: float
+    doc: The width of the wavelength range that the filter transmits or blocks (full
+      width at half maximum).
+  - name: filter_type
+    dtype: text
+    doc: Type of filter (e.g., 'Bandpass', 'Bandstop').
+  - name: model
+    dtype: text
+    doc: Model of the optical filter.
+    required: false
+- neurodata_type_def: EdgeOpticalFilter
+  neurodata_type_inc: Device
+  doc: Extends Device to hold an Edge Optical Filter (Longpass or Shortpass).
+  attributes:
+  - name: cut_wavelength_in_nm
+    dtype: float
+    doc: ' The wavelength at which the filter transmits half as much as its peak transmission.'
+  - name: filter_type
+    dtype: text
+    doc: Type of filter (e.g., 'Longpass', 'Shortpass').
+  - name: slope_in_percent_cut_wavelength
+    dtype: float
+    doc: The steepness of the transition from high blocking to high transmission (or
+      vice versa). Specified as a percentage of the cut wavelength.
+    required: false
+  - name: slope_starting_transmission_in_percent
+    dtype: float
+    doc: The percent transmission that defines the starting point for the slope (e.g.
+      10%).
+    required: false
+  - name: slope_ending_transmission_in_percent
+    dtype: float
+    doc: The percent transmission that defines the ending point for the slope (e.g.
+      80%).
+    required: false
+  - name: model
+    dtype: text
+    doc: Model of the optical filter.
+    required: false
+- neurodata_type_def: ObjectiveLens
+  neurodata_type_inc: Device
+  doc: Extends Device to hold a Objective lens.
+  attributes:
+  - name: model
+    dtype: text
+    doc: Model of the objective lens.
+    required: false
+  - name: numerical_aperture
+    dtype: float
+    doc: Numerical aperture, e.g., 0.39 NA.
+    required: false
+  - name: magnification
+    dtype: float
+    doc: Lens magnification.
+    required: false
+- neurodata_type_def: Effector
```

Comment:
Similar to the indicator, I have a hard time seeing this as a 'device' in the sense of other things that are descend from it

Perhaps an `Opsin` would be the base class of both? And it would just be a generic `Container` type or something like that?

Reply:
I would define opsin as a sub-category of effector; thus, I would not use it as a base class for both. I know they have the exact same fields, but the name here implies the specific use for it: you would use indicators when "reading" the activity of a cell. Instead, you would use an effector when "writing" the activity on a cell, meaning stimulate the cell to inhibit or excite a response. I feel like the names `Indicator` and `Effector` are already at the highest level of generalization, but I can do a little research to see if I can find a common name for both that would be recognized by the user.

Reply:
It's fine if they don't have a parent name to unify them both; I'd still separate them from the 'device' similar to the other discussion; that's my biggest issue

I would not say `isinstance(gcamp6, Device) is True`